ui/components/data-sources/DeleteDataSourceDialog.js

```diff
@@ -21,16 +22,26 @@ const DeleteDataSourceDialog = ({
     const removeDatasource = () => {
         const { name } = filter;
         const dataSourceId = dataSource.id;
+        const queriesToRefetch = [{
+            query: GetDataSources,
+            variables: { name }
+        }, {
+            query: GetDataSources,
+            variables: { name: undefined }
+        }];
+
+        if (hasResolvers) {
+            dataSource.resolvers.forEach(resolver => {
+                queriesToRefetch.push({
```

Comment:
I think this is a bit overkilling, considering the schemaId and type will be the same in many cases. I wonder if it would be possible to somehow simplify the refetch, something like
```
{
    query: GetResolvers,
    variables: { all: true }
}
```

Reply:
Or:
```
dataSource.resolvers.forEach(resolver => {
    const query = {
        query: GetResolvers,
        variables: { schemaId: resolver.GraphQLSchemaId, type: resolver.type }
    };
    if (!queriesToRefetch.indexOf(query)) {
        queriesToRefetch.push(query);
    }
});
```